src/components/GetMapTrack/GetMapTrack.tsx

```diff
@@ -1,22 +1,22 @@
-import React, { useState } from 'react'
+import React from 'react'
 import { ChallengeMap } from '../Map/ChallengeMap'
 import { UploadButton } from '../UploadButton/UploadButton'
-import { useFileContent } from '../UploadButton/useFileContent'
 import { parseGpxData } from '../../model/parser'
+import { useFiles } from '../../model/useFiles'
+import { useFilesToTextMap } from '../../model/useFileToText'
 
 const GetMapTrack = () => {
-  const [file, setFile] = useState<File | null>(null)
-  const fileContent = useFileContent(file)
+  const [files, , addFiles, clearFiles] = useFiles()
+  // file upload manipulation
+  const fileTextMap = useFilesToTextMap(files)
   return (
     <>
       <ChallengeMap
-        tracks={[
-          {
-            name: file?.name ?? 'no track selected',
-            path: parseGpxData(fileContent ?? ''),
-            color: [0, 0, 255],
-          },
-        ]}
+        tracks={[...fileTextMap.entries()].map(([file, fileText]) => ({
+          name: file.name,
+          path: parseGpxData(fileText ?? ''),
+          color: [0, 0, 255],
+        }))}
```

Comment:
This re-renders whenever the page re-renders. IE, It happens WAY too much. Whenever someone drags the map, it calculates this map a hundred times plus.

The fix: use a custom hook for track data.